delta-lake/src/main/java/org/apache/iceberg/delta/BaseSnapshotDeltaLakeTableAction.java

```diff
@@ -310,6 +351,11 @@ private DataFile buildDataFileFromAction(Action action, Table table) {
 
     FileFormat format = determineFileFormatFromPath(fullFilePath);
     InputFile file = deltaLakeFileIO.newInputFile(fullFilePath);
+    if (!file.exists()) {
+      throw new NotFoundException(
+          "The file %s does not exist in the Delta Lake table at %s",
+          fullFilePath, deltaTableLocation);
```

Comment:
I think checking file existence explicitly here can make the code logic more clear and have more reasonable error message

Reply:
nit: we can be more precise about the message, something like "File %s is referenced in Delta lake metadata but cannot be found in storage"